src/util/persistent_logger.js

```diff
@@ -105,10 +106,20 @@ class PersistentLogger {
      * @param {string} data 
      */
     async append(data) {
-        const fh = await this.init();
-
         const buf = Buffer.from(data + '\n', 'utf8');
-        await fh.write(this.fs_context, buf, buf.length);
+
+        for (let attempt = 0; attempt < APPEND_ATTEMPTS_LIMIT; ++attempt) {
+            const fh = await this.init();
+            //if another process has deleted the active file,
```

Comment:
The other process should not be able to delete the file if there is >= 1 process that exists as a writer. The reason is that the delete would happen only after `_process` would finish processing the active log file and in order to finish this the `collect_and_process` must have finished which must acquire an `EXCLUSIVE` lock before doing so.